src/systems/bit-media-frames.js

```diff
@@ -133,89 +157,152 @@ const snapToFrame = (() => {
   };
 })();
 
-function setMatrixScale(obj, scaleArray) {
+const setMatrixScale = (() => {
   const position = new THREE.Vector3();
   const quaternion = new THREE.Quaternion();
   const scale = new THREE.Vector3();
   const m4 = new THREE.Matrix4();
-  obj.updateMatrices();
-  obj.matrixWorld.decompose(position, quaternion, scale);
-  setMatrixWorld(obj, m4.compose(position, quaternion, scale.fromArray(scaleArray)));
-}
 
-function cloneForPreview(world, eid) {
-  // TODO We assume capturable object is an AFRAME entity
-  const el = world.eid2obj.get(eid).el;
-  const mesh = el.getObject3D("mesh");
-  const meshClone = cloneObject3D(mesh, false);
-  meshClone.traverse(node => {
-    updateMaterials(node, function (srcMat) {
-      const mat = srcMat.clone();
-      mat.transparent = true;
-      mat.opacity = 0.5;
-      mat.format = THREE.RGBAFormat;
-      mat.blending = THREE.NormalBlending;
-      node.material = mat;
-      return mat;
+  return (obj, scaleArray) => {
+    obj.updateMatrices();
+    obj.matrixWorld.decompose(position, quaternion, scale);
+    setMatrixWorld(obj, m4.compose(position, quaternion, scale.fromArray(scaleArray)));
+  };
+})();
+
+const frame2mixer = new Map();
+function createPreviewMesh(world, frame, capturable) {
+  let srcMesh;
+  let el;
+  let previewMesh;
+  let isVideo = false;
+  let ratio = 1;
+  if (hasComponent(world, AEntity, capturable)) {
+    el = world.eid2obj.get(capturable).el;
+    const video = el.components["media-video"];
+    isVideo = !!video;
+    if (isVideo) {
+      ratio =
+        (video.videoTexture.image.videoHeight || video.videoTexture.image.height) /
+        (video.videoTexture.image.videoWidth || video.videoTexture.image.width);
+    }
+    srcMesh = el.getObject3D("mesh");
+  } else {
+    const mediaEid = findChildWithComponent(world, MediaLoaded, capturable);
+    isVideo = hasComponent(world, MediaVideo, mediaEid);
+    if (isVideo) {
+      ratio = MediaVideo.ratio[mediaEid];
+    }
+    srcMesh = APP.world.eid2obj.get(mediaEid);
+  }
+
+  // Audios can't be cloned so we take a different path for them
+  if (isVideo) {
+    const previewMaterial = new THREE.MeshBasicMaterial();
+    previewMaterial.side = THREE.DoubleSide;
+    previewMaterial.transparent = true;
+    previewMaterial.opacity = 0.5;
+
+    const geometry = createPlaneBufferGeometry(1, 1, 1, 1, TEXTURES_FLIP_Y);
```

Comment:
I noted in the cleanup code, we would technically need to clean this up (since we are the ones creating a disposable resource), but we *don't* want to dipose of the `previewMesh`'s geometry for the other paths (since we are just referencing them. Since we don't actually mutate this geometry at all we can just statically define it once and share it among all media frames, and never `dispose()` of it.